src/pages/VenuePage/VenuePage.tsx

```diff
@@ -173,20 +263,44 @@ export const VenuePage: React.FC = () => {
 
   // useVenueAccess(venue, handleAccessDenied);
 
-  if (venueRequestStatus && !venue) {
+  if (isVenueLoaded && !venue) {
     return <>This venue does not exist</>;
   }
 
-  if (!venue || !venueId) {
-    return <LoadingPage />;
+  if (!venueId) {
+    // this should be happening only if invalid url param
+    // no sense in displaying loading page, another message or action is needed
+    // @debt use/display proper message/action when !venueId
+    return null;
+  }
+
+  if (!venue) {
+    return (
+      <Suspense fallback={<></>}>
+        <LoadingPage />
+      </Suspense>
+    );
   }
 
   if (!user) {
-    return <Login venue={venue} />;
+    return (
+      <>
+        <Preload venue={venue} />
+        <Login venue={venue} />
+      </>
+    );
   }
```

Comment:
At this stage i'm not convinced that `Preload` is going to have any meaningful impact on things; and if we do decide to keep it, it shouldn't end up duplicated in so many places.

Reply:
Superseded by single hook in `VenuePage` component e9da1d5788e915f6de13a7aa0315601017c75b73